server-test/src/server-test/buildserver/build.sbt

```diff
@@ -31,9 +31,7 @@ lazy val respondError = project.in(file("respond-error"))
     }
   )
 
-lazy val util = project.settings(
-  Compile / target := baseDirectory.value / "custom-target",
-)
+lazy val util = project
```

Comment:
I think we should keep this. Otherwise it might be too hard for some builds to migrate to sbt 2.x.

We should better fail at caching than fail at building.

Reply:
I honestly don't see a huge value in allowing `target` to be changed at the subproject level. We should keep `target` as a backward compatible key, but generally sbt 2.x generates outputs in one single directory also called `target` at the working directory.

Reply:
> I honestly don't see a huge value in allowing target to be changed at the subproject level.

Sure. But it's also about the `classDirectory`, and all the `package` files. It's tempting to move them out of the output directory, which would fail the action cache. This test is the only one that checks that the cached tasks don't fail if its output files are outside the cache directory.

I am not too much concerned about new builds. People will quickly understand they are not supposed to move output files outside the output directory. But I am more concerned about all the existing builds. We should be cautious not to break things that were permitted before and testing them.